tensorflow_model_optimization/python/examples/clustering/keras/mnist/mnist_cnn.py

```diff
@@ -43,158 +44,149 @@
+
+  # Checking that the deserialized model's accuracy matches the clustered model
+  score = loaded_clustered_model.evaluate(x_test, y_test, verbose=0)
+  print('Deserialized model test loss:', score[0])
+  print('Deserialized model test accuracy:', score[1])
+
+  # Ensure accuracy persists after stripping the model
+  stripped_model = cluster.strip_clustering(loaded_clustered_model)
+  stripped_model.compile(
+    loss=tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True),
     optimizer='adam',
     metrics=['accuracy'])
-    stripped_model.save('stripped_model.h5')
 
-    # To acquire the stripped model,
-    # deserialize with clustering scope
-    with cluster.cluster_scope():
-      loaded_model = keras.models.load_model('stripped_model.h5')
+  # Checking that the stripped model's accuracy matches the clustered model
+  score = stripped_model.evaluate(x_test, y_test, verbose=0)
+  print('Stripped model test loss:', score[0])
+  print('Stripped model test accuracy:', score[1])
 
-    # Checking that the stripped model's accuracy matches the clustered model
-    score = loaded_model.evaluate(x_test, y_test, verbose=0)
-    print('Stripped Model Test loss:', score[0])
-    print('Stripped Model Test accuracy:', score[1])
 
 def main(unused_argv):
   if FLAGS.enable_eager:
     print('Running in Eager mode.')
     tf.compat.v1.enable_eager_execution()
 
-  # input image dimensions
-  img_rows, img_cols = 28, 28
-
   # the data, shuffled and split between train and test sets
-  (x_train, y_train), (x_test, y_test) = tf.keras.datasets.mnist.load_data()
-
-  if tf.keras.backend.image_data_format() == 'channels_first':
-    x_train = x_train.reshape(x_train.shape[0], 1, img_rows, img_cols)
-    x_test = x_test.reshape(x_test.shape[0], 1, img_rows, img_cols)
-    input_shape = (1, img_rows, img_cols)
-  else:
-    x_train = x_train.reshape(x_train.shape[0], img_rows, img_cols, 1)
-    x_test = x_test.reshape(x_test.shape[0], img_rows, img_cols, 1)
-    input_shape = (img_rows, img_cols, 1)
-
-  x_train = x_train.astype('float32')
-  x_test = x_test.astype('float32')
-  x_train /= 255
-  x_test /= 255
+  (x_train, y_train), (x_test, y_test) = load_mnist_dataset()
+
   print('x_train shape:', x_train.shape)
   print(x_train.shape[0], 'train samples')
   print(x_test.shape[0], 'test samples')
 
-  # convert class vectors to binary class matrices
-  y_train = tf.keras.utils.to_categorical(y_train, num_classes)
-  y_test = tf.keras.utils.to_categorical(y_test, num_classes)
-
-  sequential_model = build_sequential_model(input_shape)
-  functional_model = build_functional_model(input_shape)
-  models = [sequential_model, functional_model]
-  train_and_save(models, x_train, y_train, x_test, y_test)
+  # Build model
+  model = build_sequential_model()
+  # Train model
+  model = train_model(model, x_train, y_train, x_test, y_test)
+  # Cluster and fine-tune model
+  clustered_model = cluster_model(model, x_train, y_train, x_test, y_test)
```

Comment:
LGTM.
This is more of a question than a comment:
Have you tried to apply the clustering wrapper on the inference graph rather than the training graph of a small example like mnist? If yes, what does the curve of the training loss look like?

Reply:
I'm not sure what you mean by "inference graph" and "training graph". Could you explain this a bit further?

Reply:
Hi, yes. An inference graph usually refers to a training graph after modifications targeted at inference. For instance, you can take a look at a TensorFlow [legacy ](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/graph_transforms/README.md)tool explaining some typical graph transforms (e.g. fold_batch_norms, merge_duplicate_nodes, remove_control_dependencies). I will write a summary on the internal site soon.

In our application context, I was just curious whether clustering behaves differently on a toy model, when tested with an inference graph rather than a training graph, in a similar way to a realistic model. If so, it could help our future debugging for upcoming new features with all the nice visualization you have done. We can check this later and this is not essential for this PR.